pkg/controller/flinkapplication/flink_state_machine.go

```diff
@@ -181,6 +190,7 @@ func (s *FlinkStateMachine) handleClusterStarting(ctx context.Context, applicati
 	// Wait for all to be running
 	ready, err := s.flinkController.IsClusterReady(ctx, application)
 	if err != nil {
+		s.getAndUpdateError(ctx, application, err)
```

Comment:
In ClusterStarting, there's a class of errors with the underlying deployments (or pods) that could prevent the cluster from coming up (for example, a bad image). I think we'll still need some kind of time limit to handle that case, as currently the deployment will just get stuck indefinitely.

Reply:
I've also included a time-based check on errors that are neither retryable nor deemed to fail fast. Let me know if that doesn't make sense.